Login Use case (For-12)

LL333111 opened the pull request:
- Half complete, I have implemented the log in use case. Not yet finish the login View, but close to finish

LL333111 committed:
- Login Use case is complete
- Some additional getters have to be implemented for AnonymousUser class
- UserFactory, User create(name, password), not enough, need to include ID,birthdate...

LL333111 committed:
- Because useID, userEmail, userBirthDate and userFull name are needed for the loginInteractor class. So additional four getter is added to the user Interface which lead to some other minor changes to other files in the entity package.

Karththigan commented on src/main/java/use_case/login/LoginInputData.java:
```diff
@@ -0,0 +1,47 @@
+    private String birthData;
+    private String fullName;
+
+    public LoginInputData(String userID, String userName, String password,
```
Shouldn't users only require a username and password to login? (no userID, birthdate or full name)?

I think all the other fields are required for signup only, not login

Pinging @izabellemarianne for opinion

LL333111 replied:
Yeah, that's right the user only need username and password to login. I included userID, birthdate and full name, because based on the UML diagram of the login use case, they need to be included. So, I’m not entirely sure if there are specific reasons for their inclusion. I think I will change it to just username and password for now and also to avoid the merge conflicts in the entity folder.

ethanliu24 replied:
Hmm I think we should login through email, since usernames could overlap? That's how we implemented sign up, we can't sign up if the entered email exists

LL333111 replied:
Oh, I thought we were going to make usernames unique. I will change it to email instead.

Karththigan commented on src/main/java/use_case/login/LoginOutputData.java:
```diff
@@ -0,0 +1,70 @@
+        this.userName = username;
+        this.loginSuccessful = loginSuccessful;
+        this.password = password;
+        this.fullname = fullname;
```
Same comment as above

ethanliu24 commented:
Hey @LL333111 I see that u made some changes to the Entity folder to make ur code work (which is good!), but izabelle is also going to make changes to that so there will be merge conflicts. So for this PR can u unstage the entity files u changed (keep ur other code the same, if no changes need to be made from the reviews). I'm going to leave it to u to figure out how to unstage files from git, some google search would help :)

Also @izabellemarianne you can look at lucas's code and see if it conflicts with ur changes (e.g. the order of the parameters passed in)

LL333111 committed:
- Change login back to only email and password.
- Delete changes made to the entity folder,now it is in the original state from Izabelle’s version.
- DBUserDataAccessObject implement LoginDataAccessInterface

LL333111 committed:
- added getCurrentUserEmail method to LoginDataAccessInterface

LL333111 committed:
,

LL333111 committed:
- Implemented the LoginView but have not yet implemented the switchToSignupView

LL333111 committed:
unstage

ethanliu24 reviewed (changes requested):
@LL333111 Please unstage/remove **all** of your files except the ones in `login` use case folder. That means, in the `Files Changed` tab, you should only see the files in `use_case/login` in there. Save ur code somewhere else so you don't have to code it again when you eventually get there. Check my message on discord or do your own research on how to unstage/remove files.

LL333111 committed:
unstage

ethanliu24 commented:
This PR is technically closed, it's kept here for reference
